lmdeploy/pytorch/backends/cuda/activation.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from torch import nn
+
+from lmdeploy.pytorch.kernels.cuda.activation import silu_and_mul
+
+from ..activation import SiluAndMulBuilder, SiluAndMulImpl
+
+
+class TritonSiluAndMulImpl(SiluAndMulImpl):
+    """silu + multiple residual fused implementation."""
+
+    def __init__(self, inplace: bool):
+        self.inplace = inplace
+
+    def _forward_naive(self, x):
+        """forward naive."""
+        gate, up = x.chunk(2, -1)
+        return nn.functional.silu(gate, self.inplace) * up
+
+    def forward(self, x):
+        """forward."""
+
+        if x.size(-1) % 2048 != 0:
```

Comment:
why only use fused kernel when `x.size(-1) % 2048 == 0`.

Reply:
I fixed block size in

https://github.com/InternLM/lmdeploy/blob/f6623327d8eeb85c024c93f20f7edbc793089595/lmdeploy/pytorch/kernels/cuda/activation.py#L62

I am so lazy.

Reply:
I didn't get it

Reply:
The kernel would be more complex and slow if we had to support an arbitrary input shape.
This kernel only support aligned input. Unaligned input would be computed using the default implementation.